packages/client/src/views/Performance/Dashboard.tsx

```diff
@@ -0,0 +1,68 @@
+/*
```

Comment:
Would it be possible to make most parts of this component reusable so that all

packages/client/src/views/Performance/Dashboard.tsx
packages/client/src/views/Performance/LeaderBoards.tsx
packages/client/src/views/Performance/Statistics.tsx

Would have the same functional logic?

This would also make 2 test suites redundant.

Reply:
when you mean reusable what i understand is this:
`
interface Dashboard {
title: string
icon: ReactWrapper
url: string
}
export function dashboardComponent(props: Dashboard) {
  const dispatch = useDispatch()
  return (
    <>
      <Frame
        header={
          <AppBar
            desktopTitle={props.title}
            desktopLeft={props.icon}
            desktopRight={
              <Button
                type="icon"
                size="medium"
                onClick={() => dispatch(goBack())}
              >
                <Icon name="X" color="primary" />
              </Button>
            }
            mobileLeft={props.icon}
            mobileRight={
              <Button
                type="icon"
                size="medium"
                onClick={() => dispatch(goBack())}
              >
                <Icon name="X" color="primary" />
              </Button>
            }
            mobileTitle={props.title}
          />
        }
        skipToContentText="Skip to main content"
      >
        {window.config && props.url ? (
          <StyledIFrame
            src={props.url}
            allowFullScreen
          />
        ) : (
          <Content title="Dashboard" size="large">
            {' '}
            No Content{' '}
          </Content>
        )}
      </Frame>
    </>
  )
}`

or am i going on the wrong path?

Reply:
Yea, exactly what I mean. The outcome should be a component that could be used like 

```ts
<DashboardEmbedView title="Leaderboard" url={window.config.LEADERBOARDS_DASHBOARD_URL} />
```

Reply:
alright i got it.